crates/matrix-sdk-base/src/rooms/mod.rs

```diff
@@ -72,42 +72,104 @@ impl BaseRoomInfo {
     /// Handle a state event for this room and update our info accordingly.
     ///
     /// Returns true if the event modified the info, false otherwise.
-    pub fn handle_state_event(&mut self, content: &AnyStateEventContent) -> bool {
-        match content {
-            AnyStateEventContent::RoomEncryption(encryption) => {
-                self.encryption = Some(encryption.clone());
+    pub fn handle_state_event(&mut self, ev: &AnySyncStateEvent) -> bool {
+        match ev {
+            // No redacted branch - enabling encryption cannot be undone.
+            AnySyncStateEvent::RoomEncryption(SyncStateEvent::Original(encryption)) => {
+                self.encryption = Some(encryption.content.clone());
             }
-            AnyStateEventContent::RoomAvatar(a) => {
-                self.avatar_url = a.url.clone();
+            AnySyncStateEvent::RoomAvatar(a) => {
+                self.avatar_url = a.as_original().and_then(|a| a.content.url.clone());
             }
-            AnyStateEventContent::RoomName(n) => {
-                self.name = n.name.as_ref().map(|n| n.to_string());
+            AnySyncStateEvent::RoomName(n) => {
+                self.name =
+                    n.as_original().and_then(|n| n.content.name.as_ref().map(|n| n.to_string()));
             }
-            AnyStateEventContent::RoomCreate(c) if self.create.is_none() => {
-                self.create = Some(c.clone());
+            // FIXME: What about redacted m.room.create events?
```

Comment:
Let's not add FIXMEs to the mainline code base. If we want this fixed, let's add an issue in the issue tracker please.